src/hooks/useDappMetadata.ts

```diff
@@ -1,22 +1,88 @@
 /* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import { CoreTypes } from "@walletconnect/types";
 import {
-  useWalletKitStore,
   WalletKitEvent,
   WalletKitEventTypes,
+  WalletKitSessionRequest,
 } from "ducks/walletKit";
 import { useMemo } from "react";
 
 /**
  * Default empty metadata object used when no metadata is available
  */
-const emptyMetadata: CoreTypes.Metadata = {
+export const emptyMetadata: CoreTypes.Metadata = {
   name: "",
   description: "",
   url: "",
   icons: [],
 };
 
+/**
+ * Pure utility to extract dApp metadata from a WalletKit event or session request.
+ */
+export function extractDappMetadata(
+  eventOrRequest: WalletKitEvent | WalletKitSessionRequest | null,
+): CoreTypes.Metadata {
+  if (!eventOrRequest) return emptyMetadata;
```

Comment:
I think it's important to return `null` here instead of `emptyMetadata` due to some checks we are doing on the Dapp bottom sheets which only works when this returns `null` (or `undefined`) instead of an object